packages/api/amplify_api_ios/ios/Classes/SwiftAmplifyApiPlugin.swift

```diff
@@ -62,33 +63,29 @@ public class SwiftAmplifyApiPlugin: NSObject, FlutterPlugin {
 
             let arguments = try FlutterApiRequest.getMap(args: callArgs)
 
-            if method == "addPlugin"{
-                let authProvidersList = arguments["authProviders"] as? [String] ?? []
-                let authProviders = authProvidersList.compactMap {
-                    AWSAuthorizationType(rawValue: $0)
-                }
-                addPlugin(authProviders: authProviders, result: result)
-                return
-            }
-
             try innerHandle(method: method, arguments: arguments, result: result)
         } catch {
             print("Failed to parse query arguments with \(error)")
             FlutterApiErrorHandler.handleApiError(error: APIError(error: error), flutterResult: result)
         }
     }
-
-    private func addPlugin(authProviders: [AWSAuthorizationType], result: FlutterResult) {
+    
+    public func addPluginAuthProvidersList(
+        _ authProvidersList: [String],
+        _ errorPtr: AutoreleasingUnsafeMutablePointer<FlutterError?>
+        ) {
```

Comment:
Formatting. 

Also, I believe the parameters must be named a particular way for ObjC interop to work correct. Please verify this works and/or delete the function and have Swift regenerate it for you.